### PR TITLE
chore: generate: improve lets/globals eval errs logs

### PR DESCRIPTION
**Pull request by katcipis**
# Reason for This Change

Improve error messages on some generate errors. Also fix a bug on the errors pkg where we were ommiting the filename from the range when the origin file is .tm.hcl.

### CODE REVIEW

File: errors/error.go [hunk 1]
```diff
@@ -463,9 +463,16 @@ func equalStack(s1, s2 StackMeta) bool {
 		s1.Path() == s2.Path()
 }
 
+// cleanFilename will clean the given filename returning a placehold if
```
Comment (i4ki) [inline errors/error.go:466]:
```suggestion
// cleanFilename will clean the given filename returning a placeholder if
```

File: errors/error.go [hunk 2]
```diff
@@ -463,9 +463,16 @@ func equalStack(s1, s2 StackMeta) bool {
...
+// it is not a valid unix/win path. This is necessary since we use some
+// hacks on partial evaluation like adding a Filename on expressions that
+// is not a valid path but that includes information required by terramate
+// embedded on them, like the expression raw bytes.
 func cleanFilename(fname string) string {
-	if ext := filepath.Ext(fname); ext == ".tm" || ext == ".tm.hcl" {
-		return fname
+	// One of the few restrictions for paths to be valid is to not
+	// have any zeroes in the middle of the string.
```
Comment (i4ki) [inline errors/error.go:473]:
```suggestion
	// have any NUL bytes in the middle of the string.
```

File: errors/error_test.go [hunk 3]
```diff
@@ -460,6 +460,48 @@ func TestErrorIs(t *testing.T) {
 	}
 }
 
+func TestErrorRangeRepresentation(t *testing.T) {
```
Comment (i4ki) [inline errors/error_test.go:463]: top

File: generate/genhcl/genhcl.go [hunk 4]
```diff
@@ -16,7 +16,7 @@
 package genhcl
 
 import (
-	stdmt "fmt"
```
Comment (i4ki) [inline generate/genhcl/genhcl.go:19]: o.o I did this? hahaha sorry

**Review by i4ki — APPROVED** [review]
lgtm

File: generate/genhcl/genhcl.go (hunk 4 above)
Comment (katcipis) [reply generate/genhcl/genhcl.go:19]: Didnt see it on the review either, brain autocompletes I suppose =P

**Review by i4ki — APPROVED** [review]
lgtm

**Pull request merged**